Libraries/DirectMLX.h

```diff
@@ -88,8 +88,8 @@ inline UINT64 DMLCalcBufferTensorSize(
     UINT64 minimumImpliedSizeInBytes = 0;
     if (!strides)
     {
-        minimumImpliedSizeInBytes = 1;
-        for (UINT i = 0; i < dimensionCount; ++i)
+        minimumImpliedSizeInBytes = sizes[0];
```

Comment:
We presume `dimensionCount` is > 0 without any check here, but that could lead to a read AV if dimensionCount is 0 (the previous code worked correctly with a scalar dimension set of {}). Granted, by the time it reaches the real API to create the operator, it must be at least size 1, but this utility function could be more general.